kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NetworkPropertiesTest.java

```diff
@@ -0,0 +1,288 @@
+		thenAnNullPointerExceptionOccured();
+	}
+
+	@Test
+	public void shouldReturnPropertiesWithGetProperties() {
+		givenTheMapWith("testKey1", "testString1");
+		givenNetworkPropsIsCreated();
+		whenGetPropertiesIsCalled();
+		thenResultEquals(Collections.singletonMap("testKey1", "testString1"));
+	}
+
+	@Test
+	public void shouldReturnStringWithGet() {
+		givenTheMapWith("testKey1", "testString1");
+		givenNetworkPropsIsCreated();
+		whenGetIsCalledWith("testKey1", String.class);
+		thenResultEquals("testString1");
+	}
+	
+	@Test
+	public void shouldReturnStringWithGetNull() {
+		givenTheMapWith("testKey1", null);
+		givenNetworkPropsIsCreated();
+		whenGetIsCalledWith("testKey1", String.class);
+		thenAnNoSuchElementExceptionOccured();
+	}
+	
+	@Test
+	public void shouldReturnStringWithGetMissing() {
+		givenTheMapWith("testKey1", null);
+		givenNetworkPropsIsCreated();
+		whenGetIsCalledWith("testKey1-nonExistant", String.class);
+		thenAnNoSuchElementExceptionOccured();
+	}
+
+	@Test
+	public void shouldReturnStringWithGetOpt() {
+		givenTheMapWith("testKey1", "testString1");
+		givenNetworkPropsIsCreated();
+		whenGetOptIsCalledWith("testKey1", String.class);
+		thenResultEquals(Optional.of("testString1"));
+	}
+
+	@Test
+	public void shouldReturnEmptyGetOptEmpty() {
+		givenTheMapWith("testKeyEmpty", "");
+		givenNetworkPropsIsCreated();
+		whenGetOptIsCalledWith("testKeyEmpty", String.class);
+		thenResultEquals(Optional.empty());
+	}
+
+	@Test
+	public void shouldReturnEmptyGetOptNull() {
+		givenTheMapWith("testKeyNull", null);
+		givenNetworkPropsIsCreated();
+		whenGetOptIsCalledWith("testKeyNull", String.class);
+		thenResultEquals(Optional.empty());
+	}
+
+	@Test
+	public void shouldReturnEmptyGetOptEmptyKey() {
+		givenTheMapWith("testKeyNull", null);
+		givenNetworkPropsIsCreated();
+		whenGetOptIsCalledWith("", String.class);
+		thenResultEquals(Optional.empty());
+	}
+
+	@Test
+	public void shouldReturnStringListGetStringListWithCommaSeperatedString() {
+		givenTheMapWith("testKey-comma-seperated", "commaSeperated1,commaSeperated2,commaSeperated3");
+		givenNetworkPropsIsCreated();
+		whenGetStringListIsCalledWith("testKey-comma-seperated");
+		thenResultEquals(Arrays.asList("commaSeperated1", "commaSeperated2", "commaSeperated3"));
+	}
+
+	@Test
+	public void shouldReturnStringListGetStringListWithCommaSeperatedStringMalformed() {
+		givenTheMapWith("testKey-comma-seperated",
+				",,   ,,,commaSeperated1, ,,,,commaSeperated2,   ,,commaSeperated3,");
+		givenNetworkPropsIsCreated();
+		whenGetStringListIsCalledWith("testKey-comma-seperated");
+		thenResultEquals(Arrays.asList("commaSeperated1", "commaSeperated2", "commaSeperated3"));
+	}
+	
+	@Test
+	public void shouldReturnStringListGetStringListWithCommaSeperatedStringNull() {
+		givenTheMapWith("testKey-comma-seperated", null);
+		givenNetworkPropsIsCreated();
+		whenGetStringListIsCalledWith("testKey-comma-seperated");
+		thenAnNoSuchElementExceptionOccured();
```

Comment:
Shouldn't this be `thenA...` and not `thenAn`?

```suggestion
		thenANoSuchElementExceptionOccured();
```

> The rule is: Use "an" before a word beginning with a vowel sound (not letter)

Reply:
I was torn between grammar and proper CamelCase.  and CamelCase won....